src/equations/compressible_euler_2d.jl

```diff
@@ -896,13 +896,37 @@ end
   #                 0  sin(α)  cos(α)   0;
   #                 0    0       0      1 ]
 
-  return SVector(u[1], 
+  return SVector(u[1],
                  c * u[2] - s * u[3],
                  s * u[2] + c * u[3],
                  u[4])
 end
 
 
+##
+# TODO: merge these with the routines above in a smart way
+@inline function rotate_solution(u, normal, tangent, equations::CompressibleEulerEquations2D)
+
+  u_tilde1 = u[1]
+  u_tilde2 = u[2] * normal[1]  + u[3] * normal[2]
+  u_tilde3 = u[2] * tangent[1] + u[3] * tangent[2]
+  u_tilde4 = u[4]
+
+  return SVector(u_tilde1, u_tilde2, u_tilde3, u_tilde4)
+end
+
+
+@inline function backrotate_flux(f_tilde, normal, tangent, equations::CompressibleEulerEquations2D)
+
+  f1 = f_tilde[1]
+  f2 = f_tilde[2] * normal[1] + f_tilde[3] * tangent[1]
+  f3 = f_tilde[2] * normal[2] + f_tilde[3] * tangent[2]
+  f4 = f_tilde[4]
+
+  return SVector(f1, f2, f3, f4)
+end
```

Comment:
Sure, the method signatures will look differently in 3D. This was just a proposal for the 2D case. If we everget unstructured curved cubes in 3D, we will probably be able to do something similar there (when passing more information to the methods).

Reply:
:+1:

Reply:
I have not thought of a good way to merge these two methods. I tried to adopt the `FluxRotated` generalized flux from the `CurvedMesh` but, unfortunately, there is an implicit assumption in its use that the left/right coordinate systems run in the same direction and have the same normal scaling. This is not true for the unstructured case

Reply:
OK, then maybe just leave it as it is and create an issue for it? It seems to be that we should be able to find a way to not repeat this rotate forward/backwards, especially since the implementations look so similar already... Maybe in the end we'll just use the "advanced" method for both unstructured and structured meshes.